application/views/themeOptions/options_core.php

```diff
@@ -6,6 +6,42 @@
     "inherited value:"
 ) . ' ' . (isset($oParentOptions['font']) ? $oParentOptions['font'] : '') . ']</option>' : '');
 
+/** @var string The html for image file dropdown options */
+$imageOptions = '';
+$optgroup = '';
+foreach ($aTemplateConfiguration['imageFileList'] as $image) {
+    // If group is different than the previous one, close the previous optgroup and open a new one
+    if ($image['group'] != $optgroup) {
```

Comment:
We have 3 times near this function in one file ? Right ?

- https://github.com/LimeSurvey/LimeSurvey/blob/dc71be6dc23f871d9cd5f47b7f6dcd326afc5796/application/views/themeOptions/options_core.php#L12
- https://github.com/LimeSurvey/LimeSurvey/blob/dc71be6dc23f871d9cd5f47b7f6dcd326afc5796/application/views/themeOptions/options_core.php#L49
- https://github.com/LimeSurvey/LimeSurvey/blob/dc71be6dc23f871d9cd5f47b7f6dcd326afc5796/application/views/themeOptions/options_core.php#L74

Why not create a variable  or an function ?

Reply:
Regarding code duplication, I believe it is not worth improving it now: What would have to be done is convert the logo and the background to the new type. Now it has been tripled because there is one for the new type, one for the logo and another for the background.

Reply:
Oh right : then when move logo and background to `$attribute['type'] == 'imagefile'` it fix the issue ?

Then why not do it now ?

Reply:
I would say out fo the scope. It would require migration of already set options... a mees on its own.

Reply:
> I would say out fo the scope. It would require migration of already set options... a mees on its own.

Ah ! Right !
Need API version update,
Maybe for develop ?

It **must** be done at a time !